### DISCUSSION TITLE
Enterprise Portal Security overrides Data Hub

### DISCUSSION THREAD

**Discussion by EliShklovsky** [pimcore/pimcore#17524] [Q&A / Support] [answered]
Hi
After Portal Engine installation and configuration the Data hub end point:
_/pimcore-graphql-webservices/{configurationname}?apikey={yourApiKey}_
 is **dead** and leads to the Portal Engine login page.
The same behavior is on Pimcore's official demo - the data hub endpoint is not working.

we tried to add on security.yaml
```
access_control:
- { path: ^/pimcore-graphql-webservices, roles: PUBLIC_ACCESS }
```
But that did not help either.

Any ideas?

**Comment by fashxp** [comment]
Hmm, for on the demo it works...

**Reply by EliShklovsky** [reply]
@fashxp , did you test  [this ](https://portal-demo.pimcore.com)demo? 


https://portal-demo.pimcore.com/pimcore-graphql-webservices/products?apikey=6332aa5e6d3d6c0be31da2a8b3442113

{
  getCar(id:81){
    id
  }
}

See my test on Postman:

![Data-Hub](https://github.com/user-attachments/assets/ca4bd5f5-e520-4ac9-9507-1b352534e34f)

And also my test on Pimcore's data hub explorer:
![Data-Hub1](https://github.com/user-attachments/assets/643fb3af-eaa3-46ad-b7f5-24f49ba166d6)

**Comment by fashxp** [comment]
You are using the wrong site/domain: [https://portal-demo.pimcore.com](https://portal-demo.pimcore.com/pimcore-graphql-webservices/products?apikey=6332aa5e6d3d6c0be31da2a8b3442113)

When you are using the domain of the portal, then it is forwarding to portal login. When you use demo.pimcore.com, then it should work, at least it does for me: 
![image](https://github.com/user-attachments/assets/9cc1acf3-68f1-474e-a7ed-982b953b7212)

**Reply by EliShklovsky** [reply]
Hi @fashxp ,
It means that any Pimcore installation cannot use the same domain to data-hub and Portal Engine and we should make another sub domain for the portal?

If for example we want to exclude the /pimcore-graphql-webservices/. and make it accessible without Portal Engine login,
can we use that kind of configuration to the data hub?

```
pimcore_portal_engine:
   custom_public_routes:
   - pimcore_openidconnect_buttoninjection_script
```

**Reply by fashxp** [reply]
need to try, could work. would need to add [this route](https://github.com/pimcore/data-hub/blob/31bf83ad5688e40cc0d394fdb45b4d867decce48/src/Resources/config/pimcore/routing.yml#L10) to `custom_public_routes` then

**Reply by EliShklovsky** [reply]
Worked like charm:

```
pimcore_portal_engine:
    custom_public_routes:
        - admin_pimcoredatahub_webservice
```

I think you should add this to the documentation.

**Reply by fashxp** [reply, marked as answer]
:+1: thx for sharing, added some additional notes here. Will be live soon: https://pimcore.com/docs/platform/Portal_Engine/Installation/
